listenbrainz/webserver/static/js/src/recommendations/RecommendationCard.test.tsx

```diff
@@ -16,25 +22,297 @@ const recommendation: Recommendation = {
+    spy.mockImplementation(() => Promise.resolve(200));
+
+    instance.submitFeedback("dislike");
+    expect(spy).toHaveBeenCalledTimes(0);
+    expect(instance.props.updateFeedback).toHaveBeenCalledTimes(0);
+    expect(wrapper.exists(".recommendation-controls")).toEqual(false);
+  });
+
+  it("does nothing if CurrentUser.authtoken is not set", async () => {
+    const wrapper = shallow<RecommendationCard>(
+      <RecommendationCard
+        {...{
+          ...props,
+          currentUser: { auth_token: undefined, name: "test" },
+          updateFeedback: jest.fn(),
+        }}
+      />
+    );
+    const instance = wrapper.instance();
+
+    const spy = jest.spyOn(instance.APIService, "submitRecommendationFeedback");
+    spy.mockImplementation(() => Promise.resolve(200));
+
+    instance.submitFeedback("love");
+    expect(spy).toHaveBeenCalledTimes(0);
+    expect(instance.props.updateFeedback).toHaveBeenCalledTimes(0);
+    expect(wrapper.exists(".recommendation-controls")).toEqual(false);
+  });
+
+  it("doesn't call updateFeedback if status code is not 200", async () => {
+    const wrapper = shallow<RecommendationCard>(
+      <RecommendationCard {...props} />
+    );
+    const instance = wrapper.instance();
+    props.updateFeedback = jest.fn();
+
+    const spy = jest.spyOn(instance.APIService, "submitRecommendationFeedback");
+    spy.mockImplementation(() => Promise.resolve(201));
+
+    instance.submitFeedback("bad_recommendation");
+
+    expect(spy).toHaveBeenCalledTimes(1);
+    expect(spy).toHaveBeenCalledWith("lalala", "yyyy", "bad_recommendation");
+
+    expect(props.updateFeedback).toHaveBeenCalledTimes(0);
+    expect(
+      wrapper.find(".recommendation-controls").childAt(0).hasClass("love")
+    ).toEqual(true);
+    expect(wrapper.find('.recommendation-controls').childAt(0).childAt(0).prop("icon").iconName).toEqual("grin-stars");
+    expect(wrapper.find('.recommendation-controls').childAt(0).childAt(2).text()).toEqual("Love");
+  });
+
+  it("calls handleError if error is returned", async () => {
+    const wrapper = shallow<RecommendationCard>(
+      <RecommendationCard {...props} />
+    );
+    const instance = wrapper.instance();
+    props.updateFeedback = jest.fn();
+    instance.handleError = jest.fn();
+
+    const spy = jest.spyOn(instance.APIService, "submitRecommendationFeedback");
+    spy.mockImplementation(() => {
+      throw new Error("error");
+    });
+
+    instance.submitFeedback("dislike");
+    expect(instance.handleError).toHaveBeenCalledTimes(1);
+    expect(instance.handleError).toHaveBeenCalledWith(
+      "Error while submitting recommendation feedback - error"
+    );
+    expect(props.updateFeedback).toHaveBeenCalledTimes(0);
+    expect(
+      wrapper.find(".recommendation-controls").childAt(0).hasClass("love")
+    ).toEqual(true);
+    expect(wrapper.find('.recommendation-controls').childAt(0).childAt(0).prop("icon").iconName).toEqual("grin-stars");
```

Comment:
Same thing here, no need to check. In cases where updateFeedback should be called, we are checking the HTML elements after explicitly changing current Feedback (since it is a func of some other script). But in cases where updateFeedback is not called or is not behaving as expected, we need not check the elements,  I think.